src/Adapter/Product/CommandHandler/UpdateCombinationPricesHandler.php

```diff
@@ -0,0 +1,89 @@
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\CommandHandler;
+
+use Combination;
+use PrestaShop\PrestaShop\Adapter\Product\Repository\CombinationRepository;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Command\UpdateCombinationPricesCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\CommandHandler\UpdateCombinationPricesHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Exception\CannotUpdateCombinationException;
+
+final class UpdateCombinationPricesHandler implements UpdateCombinationPricesHandlerInterface
+{
+    /**
+     * @var CombinationRepository
+     */
+    private $combinationRepository;
+
+    /**
+     * @param CombinationRepository $combinationRepository
+     */
+    public function __construct(CombinationRepository $combinationRepository)
+    {
+        $this->combinationRepository = $combinationRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(UpdateCombinationPricesCommand $command): void
+    {
+        $combination = $this->combinationRepository->get($command->getCombinationId());
+        $updatableProperties = $this->fillUpdatableProperties($combination, $command);
+        $this->combinationRepository->partialUpdate($combination, $updatableProperties, CannotUpdateCombinationException::FAILED_UPDATE_PRICES);
+    }
+
+    /**
+     * @param Combination $combination
+     * @param UpdateCombinationPricesCommand $command
+     *
+     * @return array<int, string>
+     */
+    private function fillUpdatableProperties(Combination $combination, UpdateCombinationPricesCommand $command): array
+    {
+        $updatableProperties = [];
+
+        if (null !== $command->getImpactOnPrice()) {
+            $combination->price = (float) (string) $command->getImpactOnPrice();
+            $updatableProperties[] = 'price';
+        }
+
+        if (null !== $command->getEcoTax()) {
+            $combination->ecotax = (float) (string) $command->getEcoTax();
+            $updatableProperties[] = 'ecotax';
+        }
+
+        if (null !== $command->getUnitPriceImpact()) {
+            $combination->unit_price_impact = (float) (string) $command->getUnitPriceImpact();
```

Comment:
Cast to string then float ? 🤔 it looks not very reliable... could we improve that? Maybe have a process for such casting, with ability to detect not-castable inputs and throw exception instead?

Reply:
It is a DecimalNumber object, we've been doing like this for a while.
other option is to leave only string cast, but then need to chagne docblock in ObjectModel property to a string to (else Phpstan will complain.)

We've had raised this question some time ago, but didn't get anything better to do

Reply:
What about a new function `toFloat()` inside `DecimalNumber`?